rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java

```diff
@@ -184,7 +185,11 @@ private void processGlobalTransactionCommit() throws SQLException {
 
         try {
             if (context.hasUndoLog()) {
-                UndoLogManager.flushUndoLogs(this);
+               if(JdbcConstants.ORACLE.equalsIgnoreCase(this.getDbType())) {
```

Comment:
Maybe some factory class should take care of returning an undo manager. Otherwise you will have this kind of logic at several places.

Reply:
io.seata.rm.datasource.undo.*;   is a wrong type